### ISSUE TITLE
Fluent-bit stops sending logs altogether if one of the outputs goes down.

### ISSUE THREAD

**Issue by Nikita128** [fluent/fluent-bit#2430]
I'm trying to make a multi-cluster system with one fluent-bit agent sending logs from TAIL input plugin to two Grafana Loki instances. I'm testing the scenario, when one of the Loki instances is down, but another one should continue receiving logs. So when I'm shutting down one of the Lokies, fluent-bit agent just stops sending logs altogether (it stops sending logs to a running Loki instance too).

Here is my configuration file:
```
[INPUT]
    Name         tail
    Tag          app
    Path         var/log/app/ka*.log
    Path_Key     filename
    storage.type filesystem

[INPUT]
    Name         tail
    Tag          app_standby
    Path         var/log/app/ka*.log
    Path_Key     filename
    storage.type filesystem
  
[Output]
    Name loki
    Match app
    Url http://localhost:13100/loki/api/v1/push
    BatchWait 1
    BatchSize 3104304
    Labels {job="varlogs",agent="fluent-bit",system="app-client"}
    
[Output]
    Name loki
    Match app_standby
    Url http://192.168.15.175:13101/loki/api/v1/push
    BatchWait 1
    BatchSize 3104304
    Labels {job="varlogs",agent="fluent-bit",system="app-client"}

```

This is some strange behavior IMO. Initially I thought that the deal was with only one input plugin being specified. I though fluent-bit somehow shares them, and can't send logs to one output if another one is down, because of a mutual intermediate log storage in the filesystem. So I specified another input plugin which is the same as the first one. I saw that fluent-bit created another temporary directory for those logs too. But this couldn't solve my problem.

Where did I go wrong? How to send logs to two outputs from one fluent-bit agent if one of the outputs is down?

**Comment by edsiper** [comment]
I "think" the problem relies on Golang Loki plugin that get stuck. Can you confirm if you have another type of output like http you can reproduce the same issue

**Comment by Nikita128** [comment]
OK, I'll try to confirm it this week and respond by then.

**Comment by Nikita128** [comment]
Sorry for very late update. I've tried sending logs to Loki and PostgreSQL simultaneously, and that's like how @edsiper said. When I'd shut down postgresql server, fluent-bit kept sending logs to Loki, but when I'd shut down Loki, fluent-bit stopped sending logs to postgresql.
Also I noticed this error message popping up when Loki was unreachable:
`level=warn caller=client.go:245 id=0 component=client host=localhost:13100 msg="error sending batch, will retry" status=-1 error="Post \"http://localhost:13100/loki/api/v1/push\": dial tcp 127.0.0.1:13100: connect: connection refused"`
The thing is there was no such error when postgresql was unreachable.
So yeah. I'll create an issue at git of Loki plugin, I guess.
Thanks!
